src/Symfony/Component/Messenger/Transport/RedisExt/Connection.php

```diff
@@ -52,8 +53,8 @@ public function __construct(array $configuration, array $connectionCredentials =
         $this->connection->connect($connectionCredentials['host'] ?? '127.0.0.1', $connectionCredentials['port'] ?? 6379);
         $this->connection->setOption(\Redis::OPT_SERIALIZER, $redisOptions['serializer'] ?? \Redis::SERIALIZER_PHP);
 
-        if (isset($connectionCredentials['auth'])) {
-            $this->connection->auth($connectionCredentials['auth']);
+        if (isset($connectionCredentials['auth']) && !$this->connection->auth($connectionCredentials['auth'])) {
+            throw new InvalidArgumentException(sprintf('Redis connection failed: %s', $redis->getLastError()));
```

Comment:
@chalasr mostlywe did:

```
if ($error = $this->connection->getLastError() ?: null) {
    $this->connection->clearLastError();
}
```

Should we do the same here?

Reply:
@alexander-schranz I don't think so because we don't throw a `TransportException` here but a logic one, it is not meant to be caught by the worker

Reply:
@chalasr Sounds reasonable 👍 . Thank you for your fix!